cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/preview/SparkDataTracer.java

```diff
@@ -64,6 +64,11 @@ public void writeExternal(ObjectOutput out) throws IOException {
     out.writeUTF(dataTracer.getName());
   }
 
+  @Override
+  public int getMaximumTracedRecords() {
+    return 0;
```

Comment:
this shouldnt be returning 0, instead delegate to dataTracer instance in the class.